src/api.rs

```diff
@@ -514,22 +516,155 @@ impl Config {
   }
 }
 
+// The set of options that controls frame re-ordering and reference picture
+//  selection.
+// The options stored here are invariant over the whole encode.
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct InterConfig {
+  // Whether frame re-ordering is enabled.
+  reorder: bool,
+  // Whether P-frames can use multiple references.
+  pub(crate) multiref: bool,
+  // The depth of the re-ordering pyramid.
+  // The current code cannot support values larger than 2.
+  pub(crate) pyramid_depth: u64,
+  // Number of input frames in group.
+  pub(crate) group_input_len: u64,
+  // Number of output frames in group.
+  // This includes both hidden frames and "show existing frame" frames.
+  group_output_len: u64,
+}
+
+impl InterConfig {
+  fn new(enc_config: &EncoderConfig) -> InterConfig {
+    let reorder = !enc_config.low_latency;
+    let pyramid_depth;
+    let group_input_len;
+    let group_output_len;
+    if reorder {
+      // A group always starts with (group_output_len - group_input_len) hidden
+      //  frames, followed by group_input_len shown frames.
+      // The shown frames iterate over the input frames in order, with frames
+      //  already encoded as hidden frames now displayed with Show Existing
+      //  Frame.
+      // For example, for a pryamid depth of 2, the group is as follows:
+      //                      |TU         |TU |TU |TU
```

Comment:
Nit: (this was already in but might as well fix it now) typo on `pryamid`...also can the acronym `TU` be expanded somewhere here?

Reply:
Done.